tools/git_utils.py

```diff
@@ -62,17 +62,38 @@ def get_latest_run(workflow: Workflow) -> WorkflowRun:
     raise RuntimeError("Unable to list all workflow invocations")
 
 
-def get_name_and_commit_hash(branch_name: Optional[str]) -> tuple[str, str]:
-    branch_name = branch_name or "HEAD"
-    name = (
-        subprocess.check_output(["git", "rev-parse", "--abbrev-ref", branch_name], stderr=subprocess.STDOUT)
+def get_name_and_commit_hash(local_branch_name: Optional[str]) -> tuple[str, str]:
+    local_branch_name = local_branch_name or "HEAD"
+    has_upstream = False
+
+    try:
+        # Check if the branch has a remote tracking branch.
+        upstream_branch = (
+            subprocess.check_output(
+                ["git", "rev-parse", "--abbrev-ref", f"{local_branch_name}@{{upstream}}"], stderr=subprocess.STDOUT
+            )
+            .strip()
+            .decode("utf-8")
+        )
+        has_upstream = True
+        local_branch_name = upstream_branch
+    except subprocess.CalledProcessError:
+        pass
+
+    remote_branch_name = (
+        subprocess.check_output(["git", "rev-parse", "--abbrev-ref", local_branch_name], stderr=subprocess.STDOUT)
         .strip()
         .decode("utf-8")
     )
+    if has_upstream:
+        # Strip the upstream name from the branch to get the remote branch name.
+        remote_branch_name = remote_branch_name.split("/", 1)[1]
     commit_hash = (
-        subprocess.check_output(["git", "rev-parse", branch_name], stderr=subprocess.STDOUT).strip().decode("utf-8")
+        subprocess.check_output(["git", "rev-parse", local_branch_name], stderr=subprocess.STDOUT)
```

Comment:
This should be `remote_branch_name` then, no?

Reply:
I don't think it matters. If I'm not wrong, the commit hash should be the same for local and remote